lib/Sema/ConstraintSystem.cpp

```diff
@@ -3455,7 +3455,14 @@ void constraints::simplifyLocator(ASTNode &anchor,
 
       if (auto subscriptExpr = getAsExpr<SubscriptExpr>(anchor)) {
         anchor = subscriptExpr->getIndex();
-        path = path.slice(1);
+        // TODO: It would be better if the index expression was always wrapped
+        // in a ParenExpr.
+        if (!(isExpr<TupleExpr>(anchor) && isExpr<ParenExpr>(anchor)) &&
```

Comment:
`anchor` cannot be `TupleExpr` and `ParenExpr` at the same time, you probably meant `||` here. Also I think it's okay to do one slice right when index is extracted, you just need to check for "next" element in that case...

Reply:
Oops! Fixed now.